### PR TITLE
feature: patch a batch of nodes

### PR DESCRIPTION
**Pull request by sawall**

### CODE REVIEW

File: scripts/patch-nodes.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+fi
+
+if [ -z "$ANSIBLE_INVENTORY" ]; then 
+  ANSIBLE_INVENTORY="./batch-${ROLE}-${ORACLE_REGION}.inventory"
+  $LOCAL_PATH/node.py --environment $ENVIRONMENT --role $ROLE --region $ORACLE_REGION --oracle --batch --inventory $RELEASE_PARAM > $ANSIBLE_INVENTORY
+fi
+
+DEPLOY_TAGS=${ANSIBLE_TAGS-"all"}
+ANSIBLE_PLAYBOOK_FILE=${ANSIBLE_PLAYBOOK_FILE-"patch-nodes-default.yml"}
+ANSIBLE_PLAYBOOK="$LOCAL_PATH/../../infra-configuration/ansible/$ANSIBLE_PLAYBOOK_FILE"
```
Comment (aaronkvanmeerten) [inline scripts/patch-nodes.sh:41]: this should be $LOCAL_PATH/../ansible/$ANSIBLE_PLAYBOOK_FILE

File: ansible/patch-nodes-default.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  vars_files:
+    - secrets/ssh-users.yml
+    - secrets/ssl-certificates.yml
+    - roles/jitsi-repo/defaults/main.yml
+    - secrets/repo.yml
+    - config/vars.yml
+    - sites/{{ hcv_environment }}/vars.yml
+
+  pre_tasks:
+    - apt_repository: >
```
Comment (aaronkvanmeerten) [inline ansible/patch-nodes-default.yml:16]: The repo related pre-tasks aren't needed any longer.

File: ansible/patch-nodes-default.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+---
+- hosts: all
```
Comment (aaronkvanmeerten) [inline ansible/patch-nodes-default.yml:2]: proper linting means playbooks all of a name, so I usually do name: Main
```suggestion
- name: Main
  hosts: all
```

File: scripts/patch-nodes.sh [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            timeout 10 ssh -o StrictHostKeyChecking=no -F $LOCAL_PATH/../config/ssh.config $ANSIBLE_SSH_USER@$ip "uptime > /dev/null" && echo $ip >> ./batch.inventory || FAILED_COUNT=$(($FAILED_COUNT+1))
+        done
+    fi
+
+    LIVE_COUNT=$(cat ./batch.inventory | wc -l | awk '{print $1}')
+    if [[ $LIVE_COUNT -gt 1 ]]; then
+
+        ansible-playbook $ANSIBLE_PLAYBOOK \
+            -i ./batch.inventory \
+            -e "ansible_ssh_user=$ANSIBLE_SSH_USER hcv_environment=$ENVIRONMENT shard_role=$ROLE patch_ansible_roles=\"$ANSIBLE_ROLES\"" \
```
Comment (aaronkvanmeerten) [inline scripts/patch-nodes.sh:70]: maybe we should have an additional variable like $ANSIBLE_ADDITIONAL_VARS to allow more variables to be passed in

File: ansible/patch-nodes-default.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      tags: setup
+    - file: path=/etc/apt/auth.conf.d state=directory
+    - template: dest="/etc/apt/auth.conf.d/jitsi-repo.conf" src="roles/jitsi-repo/templates/jitsi-repo.conf.j2" owner=root mode=0600
+      tags: setup
+    - apt_repository: >
+        repo='deb [trusted=yes] {{jitsi_repo_url}} unstable/'
+        state=present
+        update_cache=no
+      tags: setup
+    - apt: update_cache=yes cache_valid_time=86400
```
Comment (aaronkvanmeerten) [inline ansible/patch-nodes-default.yml:34]: Lint requires a name for each task
Lint requires the FQCN like ansible.builtin.apt
Lint requires not using inline
```suggestion
    - name: Update apt cache
      apt:
        update_cache: true
        cache_valid_time: 86400
```

File: ansible/patch-nodes-default.yml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      tags: setup
+
+    - ansible.builtin.include_role:
```
Comment (aaronkvanmeerten) [inline ansible/patch-nodes-default.yml:37]: lint requires a name for the task
```suggestion
    - name: Include roles from patch_ansible_roles 
      ansible.builtin.include_role:
```

**Review by aaronkvanmeerten — CHANGES_REQUESTED** [review]

**Review by aaronkvanmeerten — APPROVED** [review]

File: scripts/patch-nodes.sh (hunk 4 above)
Comment (sawall) [reply scripts/patch-nodes.sh:77]: implemented as ANSIBLE_EXTRA_VARS

**Pull request merged**